components/zip321/src/lib.rs

```diff
@@ -219,563 +225,70 @@ impl Payment {
 /// payment value in the request.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct TransactionRequest {
-    payments: BTreeMap<usize, Payment>,
+    inner: zip321_parse::TransactionRequest<Payment>,
 }
```

Comment:
Much like `zip321::Payment`, `zip321::TransactionRequest` is a smart pointer to a parse `zip321_parse::TransactionRequest<zip321::Payment>`.